### PR TITLE
Fix `overrides.extends` order when including same rules

### LINKED ISSUE #6658: Fix `overrides.extends` order when including same rules
### What steps are needed to reproduce the bug?

#6380 changes the behaviour of `overrides.extends` @jasikpark

In following config file, `./a` is the last item, it should in the last of the merge array.

But in the pull request

https://github.com/stylelint/stylelint/blob/4daef246942587560baa60177985b0ae6384738f/lib/augmentConfig.js#L303-L305

`./a` in first `extends` is used.

### What Stylelint configuration is needed to reproduce the bug?

```
modules.export = {
  extends: ["./a", "./b"],
  overrides: [
    {
      files: ["*.module.css"],
      extends: ["./c", "./b", "./a"]
    }
  ]
}
```

### How did you run Stylelint?

none

### Which version of Stylelint are you using?

`15.1.0`

### What did you expect to happen?

`overrides.extends` work as expect

### What actually happened?

see previous

### Does the bug relate to non-standard syntax?

_No response_

### Proposal to fix the bug

_No response_

### PR DESCRIPTION
**Pull request by kuoruan**
> Which issue, if any, is this issue related to?

Closes #6658

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: .changeset/thin-candles-cross.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fix overrides.extends order when including same rules
```
Comment (ybiquitous) [inline .changeset/thin-candles-cross.md:5]: [suggest] According to [our convention](https://stylelint.io/maintainer-guide/pull-requests#merging):

```suggestion
Fixed: `overrides.extends` order when including same rules
```

**Review by ybiquitous — APPROVED** [review]
Thanks, LGTM 👍🏼 
In addition, please fix the changeset I've suggested.

File: .changeset/thin-candles-cross.md (hunk 1 above)
Comment (kuoruan) [reply .changeset/thin-candles-cross.md]: @ybiquitous done.

**Review by ybiquitous — APPROVED** [review]
Thank you!

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you.

**Pull request merged**